limiter/request_limiter.go

```diff
@@ -0,0 +1,146 @@
+package limiter
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"go.uber.org/zap"
+	yaml "gopkg.in/yaml.v2"
+)
+
+//Rule is a request blocking rule
+type Rule map[string]string
+
+//RuleConfig represents the request blocking rules
+type RuleConfig struct {
+	Rules []Rule
+}
+
+//RequestLimiter blocks request according to rules that defines which headers are not allowed
+type RequestLimiter struct {
+	config       iConfigFile
+	logger       *zap.Logger
+	rules        atomic.Value
+	updatePeriod time.Duration
+}
+
+//NewRequestLimiter creates a new instance of request limiter without any rules
+//and sets name of config files that will be used as storage for rules
+func NewRequestLimiter(blockHeaderFile string, updatePeriod time.Duration, logger *zap.Logger) *RequestLimiter {
+	instance := &RequestLimiter{
```

Comment:
Does adding a separate object for limiter add any value? This does not solve the global vars problem because an instance is saved in App anyway.

Reply:
It does since fields inside object enclosed by package and not accessible outside of blocker package due to naming. Main purpose of this object is to manage access to rule values. Also was possible to define variables on level of package. But we need to initialise some of them before using this package(config file name, update period). Introduction of some sort Initialisation function could solve that and instead introduced implicit time restriction between call of Initialisation function and functions that manages rules. I followed explicit path there.